software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java

```diff
@@ -395,12 +390,12 @@ public void joinCluster(String nodeName) {
             if (!hasJoinedCluster()) {
                 ScriptHelper joinClusterScript = newScript("joinCluster")
                         .body.append(sudo(format("%s cluster join %s", getRiakAdminCmd(), nodeName)))
+                        .body.append(sudo(format("%s cluster plan", getRiakAdminCmd())))
+                        .body.append(sudo(format("%s cluster commit", getRiakAdminCmd())))
```

Comment:
Very good note.
The point here is that this clusterCommit() method is made as effector.
If we go to the use case of that command it is as following: executing one of the following effectors: leaveCluster(), joinCluster(), removeNode() and maybe recoverFailedNode()
So let's say we want to support executing these commands and then invoking commitCluster() at the end.

However when setting up a new cluster blueprint it will lead that at the end of blueprint initialization the user should manually go to the UI and click on commitCluster()
The way we could workaround this is by introducing new version of the methods leaveCluster(), joinCluster(), removeNode() like leaveClusterEffector(), joinClusterEffector(), removeNodeEffector() which is without invoking commitCluster() and leave the current methods as they are.
That way deploying and operations queue would work but then it won't be very well supported in the UI, it will be easy for the user to miss to invoke commitCluster().